backend/equipmentposts/tests.py

```diff
@@ -0,0 +1,105 @@
+
+        self.in_turkey_ads = EquipmentSerializer(EquipmentPost.objects.filter(
+            Q(location="levent") | Q(location="etiler")), many=True).data
+
+        self.ads_created_by_user = EquipmentSerializer(EquipmentPost.objects.all(), many=True).data
+        self.creation_date_today = EquipmentSerializer(EquipmentPost.objects.all(), many=True).data
+
+        self.skill_between_2_4 = self.football_ads
+
+    def test_filter_by_query(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'query': 'basketball'}, HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.basketball_ads)
+
+    def test_filter_by_creation_date(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'min_creation_date': (datetime.today() - timedelta(3)),
+                                                               'max_creation_date': (datetime.today() + timedelta(1))},
+                                   HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.creation_date_today)
+
+    def test_filter_by_location(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'location': 'le'},
+                                   HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.in_turkey_ads)
+
+    def test_filter_by_sport(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'sport': 'Basketball'},
+                                   HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.basketball_ads)
+
+    def test_filter_by_owner(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'owner_id': self.user.id},
+                                   HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.ads_created_by_user)
+
+    def test_filter_by_coordinates(self):
+        response = self.client.get(reverse('equipmentpost-list'), {'min_latitude': 36.23763062438484,
+                                                               'max_latitude': 42.01901802424485,
+                                                               'min_longitude': 26.732105369671633,
+                                                               'max_longitude': 44.3513027746188},
+                                   HTTP_AUTHORIZATION=f'JWT {self.token}')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['results'], self.in_turkey_ads)
+
+    def test_filter_by_skill(self):
```

Comment:
Do we need to filter equipments by skill level? What is the logic behind it?

Reply:
If you ask my opinion, we don't. However, our class diagram for equipment posts includes a skill level field, so I thought if we are going to have a skill level field, it wouldn't hurt to be able to filter by it.